app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/ImageLoader.kt

```diff
@@ -91,13 +107,23 @@ class ImageLoader @Inject constructor(
      *
      * @return Query result.
      */
-    private fun querySHA1(SHA1: String): Boolean {
+    private fun querySHA1(SHA1: String): Boolean? {
```

Comment:
do we need to set this as nullable (same for line 78 & 114)?

Reply:
I'm using null to notify that the result was not available or it is invalid.

Should I declare some constants for this result instead of using a boolean as we want 3 states?

Reply:
How about making a Sealed Class instead?

Reply:
Done. Thanks.